src/poetry/core/pyproject/toml.py

```diff
@@ -28,8 +28,18 @@ def data(self) -> dict[str, Any]:
             if not self.path.exists():
                 self._data = {}
             else:
-                with self.path.open("rb") as f:
-                    self._data = tomllib.load(f)
+                try:
+                    with self.path.open("rb") as f:
+                        self._data = tomllib.load(f)
+                except tomllib.TOMLDecodeError as e:
+                    from poetry.core.pyproject.exceptions import PyProjectException
+
+                    msg = f"{self._path.as_posix()} is not a valid TOML file. {e}"
+
+                    if e.args[0].startswith("Cannot overwrite a value"):
+                        msg += "\nThis is often caused by a duplicate entry"
```

Comment:
```suggestion
                    if str(e).startswith("Cannot overwrite a value"):
                        msg += "\nThis is often caused by a duplicate entry."
```

`str` should be more robust. It is unlikely but `args` could be empty.